packages/components/src/components/ui/FnDocumentation.tsx

```diff
@@ -2,25 +2,65 @@ import { clsx } from "clsx";
 import React, { FC, PropsWithChildren } from "react";
 import ReactMarkdown from "react-markdown";
 
-import { type FnDocumentation as FnDocumentationType } from "@quri/squiggle-lang";
+import {
+  FnDefinition,
+  type FnDocumentation as FnDocumentationType,
+} from "@quri/squiggle-lang";
 
 import { SQUIGGLE_DOCS_URL } from "../../lib/constants.js";
 
 const Section: FC<PropsWithChildren> = ({ children }) => (
   <div className={clsx("px-4 py-2")}>{children}</div>
 );
 
+//I'm not sure if this is worth it here. Much of the input data is hidden from us at this point. It might be better to just go back to strings, or to formally parse it after having it as a string.
+const StyleDefinition: FC<{ fullName: string; def: FnDefinition }> = ({
```

Comment:
Can we use `SqLambdaSignature` instead of this?

Changes that it will require are not that tedious, I think:
- change `type SqLambdaSignature = SqLambdaParameter[]` to `type SqLambdaSignature = { inputs: SqLambdaParameter[], output: SqLambdaOutput }`
- move the `isOptional` check to the code that generates the `typeName`, in `lambdaToSqLambdaSignatures`
- in `<FnDocumentation>`, obtain the builtin lambda with `SqLambda.createFromStdlibName`, which we already have (this means that `<LambdaWidget>` will convert the function to the documentation and then `<FnDocumentation>` will convert it back...)

I hope this will work and it will allow us not to expose raw `definitions`; if I missed something then I don't mind taking this shortcut, for now, but I'd prefer to find a way to avoid it.

Reply:
I'm going to hold off on this until https://github.com/quantified-uncertainty/squiggle/pull/2694